records/2022-11-04.reliability.sms-limits.md

```diff
@@ -0,0 +1,58 @@
+# SMS Limits
+
+Date: 2022-11-04
+
+## Status
+
+**DRAFT**
+
+## Context
+
+Originally Notify only had a "combined" email and SMS limit. 
+
+These proposed changes were prompted by an incident where we had a very high and unexpected volume of SMS sends.
+
+## Implementation overview
+
+In order to enforce a new SMS-specific limit, we have added checks to the "front door" of the API, where the post requests come in that result in an SMS send:
+- for the `/<uuid:service_id>/send-notification` endpoint used by the Admin "send one off" flow: [app/service/send_notification.py](https://github.com/cds-snc/notification-api/blob/d3412ad3697162550d3f5d73958f488a305d3cf7/app/service/send_notification.py#L66)
+- for the job creation endpoint used by the Admin CSV send flow: [app/job/rest.py](https://github.com/cds-snc/notification-api/blob/d3412ad3697162550d3f5d73958f488a305d3cf7/app/job/rest.py#L149)
+- for the `/bulk` API endpoint defined in [app/v2/notifications/post_notifications.py](https://github.com/cds-snc/notification-api/blob/d3412ad3697162550d3f5d73958f488a305d3cf7/app/v2/notifications/post_notifications.py#L186)
+- for the `/sms` API endpoint defined in [app/v2/notifications/post_notifications.py](https://github.com/cds-snc/notification-api/blob/d3412ad3697162550d3f5d73958f488a305d3cf7/app/v2/notifications/post_notifications.py#L228)
+
+Before arriving at this solution, we tried adding checks to other places in the API code (ie celery), including where the notifications are created in the database and where they are sent to SES. These other locations proved prone to errors and difficult to test, which led us to move all related code to the "front door".
+
+(Placeholder for diagram)
+
```

Comment:
Can we address how we want to tackle the schedule feature with the sms limits as well, as we'd want to increase the daily limit when a new schedule comes in?

Reply:
Sounds good, we will add a new section for that.

Reply:
The new section is in "Future Work" below